File: GTK-Hello-World/c/gtk-hello-world.c
#include<gtk/gtk.h>

void button_clicked(GtkWidget *button, gpointer data)
{
    gtk_button_set_label(GTK_BUTTON(button), "Thank you");
}

void setup_view(GtkWidget *window)
{
    GtkWidget *button = gtk_button_new_with_label("Hello, World");
    g_signal_connect(G_OBJECT(button), "clicked",
        G_CALLBACK(button_clicked), NULL);

    GtkWidget *hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
    gtk_box_pack_start(GTK_BOX(hbox), button, TRUE, FALSE, 0);

    GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
    gtk_box_pack_start(GTK_BOX(vbox), hbox, TRUE, FALSE, 0);

    gtk_container_add(GTK_CONTAINER(window), vbox);
}

GtkWidget* setup_window()
{
    GtkWidget *window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
    gtk_window_set_title(GTK_WINDOW(window), "GTK Hello World");
    gtk_window_set_default_size(GTK_WINDOW(window), 400, 300);
    g_signal_connect(window, "destroy",
        G_CALLBACK(gtk_main_quit), NULL);

    setup_view(window);

    return window;
}

int main(int argc, char **argv)
{
    gtk_init(&argc, &argv);

    GtkWidget *window = setup_window();
    gtk_widget_show_all(window);

    gtk_main();
    return 0;
}

File: GTK-Hello-World/python/gtk-hello-world.py
#!/usr/bin/env python3

import gi

gi.require_version("Gtk", "3.0")
from gi.repository import Gtk  # noqa: E402


class GtkHelloWorld(Gtk.Window):
    def __init__(self):
        Gtk.Window.__init__(self, title="GTK Hello World")
        self.set_default_size(400, 300)
        self.connect("destroy", Gtk.main_quit)

        self.setup_view()

    def setup_view(self):
        self.button = Gtk.Button(label="Hello, World")
        self.button.connect("clicked", self.on_button_clicked)

        hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL)
        hbox.pack_start(self.button, True, False, 0)

        vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
        vbox.pack_start(hbox, True, False, 0)

        self.add(vbox)

    def on_button_clicked(self, button):
        button.set_label("Thank you")


if __name__ == "__main__":
    window = GtkHelloWorld()
    window.show_all()

    Gtk.main()